x/crosschain/keeper/evm_hooks.go

```diff
@@ -72,12 +74,25 @@ func (k Keeper) ProcessLogs(ctx sdk.Context, logs []*ethtypes.Log, emittingContr
 				return err
 			}
 		}
+		// We were able to parse the ZRC20 withdrawal event. However, we were unable to process it as the information was incorrect
+		// This means that there are some funds locked in the contract which cannot have an outbound , this can be a candidate for a refund
+		// TODO : Consider returning error or auto refunding the funds to the user
+		if eventWithdrawal != nil {
+			ctx.Logger().Error(fmt.Sprintf("Error processing ZRC20 withdrawal event , from Address: %s m ,to : %s,value %s,gasfee %s, protocolfee %s, err %s",
+				eventWithdrawal.From.Hex(), string(eventWithdrawal.To), eventWithdrawal.Value.String(), eventWithdrawal.Gasfee.String(), eventWithdrawal.ProtocolFlatFee.String(), err.Error()))
+		}
+
 		eZeta, err := ParseZetaSentEvent(*log, connectorZEVMAddr)
 		if err == nil {
 			if err := k.ProcessZetaSentEvent(ctx, eZeta, emittingContract, txOrigin); err != nil {
 				return err
 			}
 		}
+		// We were able to parse the ZetaSent event. However, we were unable to process it as the information was incorrect
+		if eZeta != nil {
```

Comment:
same else if?

Reply:
Remove the check, returning nil for the address mismatch event. This check is useless if we do that